### ISSUE TITLE
Path conflict on theories/homotopy_theory/path.v with core ssreflect path.v

### ISSUE THREAD

**Issue by MSoegtropIMC** [math-comp/analysis#1425]
Recently the file

https://github.com/math-comp/analysis/commits/master/theories/homotopy_theory/path.v

was introduced. This produces a path conflict:
```
.../CP.2024.10.1~8.20~2025.01/lib/coq/user-contrib/mathcomp/analysis/homotopy_theory/path.vo and
.../CP.2024.10.1~8.20~2025.01/lib/coq/user-contrib/mathcomp/ssreflect/path.vo
```
e.g. in `Finmap.v` which contains
```
From mathcomp Require Import choice path finset finfun fintype bigop.
```
This has the effect that one can compile finmap only before mathcomp-analysis, but not after it. Maybe mathcomp-analysis depends on it, so that this cannot happen, but it is still bad and might break existing code.

I would suggest to rename this file.

**Comment by affeldt-aist** [comment]
Yes, we already identified the issue https://github.com/math-comp/analysis/issues/1412
This PR to finmap https://github.com/math-comp/finmap/pull/114 is partly addressing the issue
by removing the explicit `Require Import` to `path`.
But any way the true solution in the current state is indeed to rename the file.

**Comment by MSoegtropIMC** [comment]
The other option is to write in FinMap:
```
From mathcomp.ssreflect Require Import ssreflect ssrbool ssrnat eqtype ssrfun seq.
From mathcomp.ssreflect Require Import choice path finset finfun fintype bigop.
```

**Comment by MSoegtropIMC** [comment]
And sorry for not searching the issues first.

**Comment by MSoegtropIMC** [comment]
IMHO it should be discussed in Zulip how mathcomp wants to handle this. Either one says `From mathcomp Require` is recommended - then duplicate files should be avoided, and this should be tested for.

Or one recommends to use a more specific require like `From mathcomp.ssreflect Require` - that is at least specify the top level module under mathcomp, so that file names need only be unique per module.

I would think long term the second solution is more feasible, but both are fine.

**Comment by CohenCyril** [comment]
> IMHO it should be discussed in Zulip how mathcomp wants to handle this. Either one says From mathcomp Require is recommended - then duplicate files should be avoided, and this should be tested for.

This :point_up:
